### ISSUE TITLE
Delete mutation fails if schema includes sort keys

### ISSUE THREAD

**Issue by KayIlory** [aws-amplify/amplify-android#838]
My schema includes an `@key`, and one of the fields is of type `AWSDateTime!`:

```graphql
type TestResult @model
@key(fields:["id", "date", "createDate"])
{
    id: ID!
    userID: ID!
    testResultTypeID: ID!
    testResultType: [TestResultType] @connection(keyName: "byTestResultType", fields: ["testResultTypeID"])
    testResultImage: [TestResultImage] @connection(keyName: "byTestResult", fields: ["id"])
    date: AWSDateTime!
    notes: String
    createDate: AWSDateTime!
}
```
When I do:
```java
Amplify.API.mutate(
    ModelMutation.delete(testResult),
    response -> { /* logging */ },
    error -> { /* logging */ }
);
```

I get an error in the returned GraphQL response:
> GraphQLResponse.Error{message='Variable 'input' has coerced Null value for NonNull type 'AWSDateTime!'', locations='[GraphQLLocation{line='1', column='27'}]', path='null', extensions='null'}

However, if I _comment out the `@key` line_:

```graphql
type TestResult @model
# @key(fields:["id", "date", "createDate"])
{
    id: ID!
    userID: ID!
    testResultTypeID: ID!
    testResultType: [TestResultType] @connection(keyName: "byTestResultType", fields: ["testResultTypeID"])
    testResultImage: [TestResultImage] @connection(keyName: "byTestResult", fields: ["id"])
    date: AWSDateTime!
    notes: String
    createDate: AWSDateTime!
}
```
The same mutation will work.

--------------

The `DeleteTestResultInput` generated for this `@model` is:
```graphql
input DeleteTestResultInput {
  id: ID!
  date: AWSDateTime!
  createDate: AWSDateTime!
}
```
And the problem is the presence of the `AWSDateTime!` field(s).

However, this one (without them) will work:

```graphql
input DeleteTestResultInput {
  id: ID!
}
```

**Comment by hellosagar** [comment]
yeah we need to issue to be fixed ASAP

**Comment by KayIlory** [comment]
![image](https://user-images.githubusercontent.com/51476926/93446580-451f1000-f8c9-11ea-9a58-734a8dadd9e1.png)
![image](https://user-images.githubusercontent.com/51476926/93446617-4ea87800-f8c9-11ea-910f-e3efdb880e40.png)

@jamesonwilliams  @richardmcclellan  it seems the issue is with all sort directives and not just DateTime. I tried to be smart and use an Int sortKey to see whether it was issues with serialization of dateTime. It seems this is not the case. The AppSync looked right with the DeleteTestResultInput.

**Comment by jamesonwilliams** [comment]
@richardmcclellan confirmed that the sort key will currently cause the deletion to fail.

**Comment by richardmcclellan** [comment]
To summarize the issue here, `Amplify.API` does not currently have first class support for models with the @key directive.     When using the @key directive, the delete operation has several new required parameters, which the request builders in the library don't know about, so it tries to send the delete mutation request without these required parameters, and the request fails.

As a workaround to this, until we add first-class support, you can construct the document String and variables map yourself, like this:
```
val document = """
mutation DeleteTestResult {
    deleteTestResult(${"$"}input: DeleteTestResultInput!) {
        createDate
        date
        id
        notes
        testResultTypeID
        userID
    }
}
"""

val input = mapOf(
    "id" to "id_of_item_to_delete",
    "date" to Temporal.DateTime("1970-01-01T00:00:00Z"),
    "createDate" to Temporal.DateTime("1970-01-01T00:00:00Z"))

val variables = mapOf("input" to input)

val request: SimpleGraphQLRequest<TestResult> = SimpleGraphQLRequest(
        document,
        variables,
        TestResult::class.java,
        GsonVariablesSerializer()
)
Amplify.API.mutate(request,
    { Log.i("MainActivity", "mutation succeeded: ${it}") },
    { Log.e("MainActivity","delete failed: ${it.message}", it) }
)
```

**Comment by richardmcclellan** [comment]
This was fixed in [1.17.5](https://github.com/aws-amplify/amplify-android/releases/tag/release_v1.17.5)!
